numpy/ma/core.py

```diff
@@ -1430,7 +1427,7 @@ def getmask(a):
 get_mask = getmask
 
 
-def getmaskarray(arr):
+def getmaskarray(arr, nomask_as_readonly=False):
```

Comment:
Flipping the call stack around sounds sensible. Do you still think there should be an argument here to expose the behaviour publicly?

> I'd also suggest to make the argument `writable=True`

I went for `readonly` because that's consistent with `np.lib.stride_tricks._broadcast_to`, but I guess you're right, making the name match the flags makes more sense.